packages/chevrotain/src/parse/parser/smart_parser/smart_parser.ts

```diff
@@ -0,0 +1,82 @@
+import {
+    IRuleConfig,
+    IProduction,
+    TokenVocabulary,
+    IParserConfig
+} from "../../../../api"
+import { Parser, DEFAULT_PARSER_CONFIG } from "../parser"
+import {
+    PROP_ANALYSER_CONTEXT,
+    ParserRuleAnalyserContext
+} from "./rule-analyser"
+import { MixedInParser } from "../traits/parser_traits"
+import { keys, applyMixins, cloneObj } from "../../../utils/utils"
+import { Rule } from "../../grammar/gast/gast_public"
+import { SmartRecognizerEngine } from "./smart_recognizer_engine"
+import { SmartRecognizerApi } from "./smart_recognizer_api"
+
+export class SmartParser extends Parser {
```

Comment:
Should not this be just another trait? 

Meaning:
- Same Folder as the other traits
- Same pattern (not directly extending "Parser")